tests/plugins/clipboard/dropimage.js

```diff
@@ -88,9 +107,51 @@
 			FileReader.setReadResult( 'load' );
 
 			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
-			assertDropImage( this.editor, dropEvt, imageType, expected, {
-				dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
-				dropOffset: 17
+			assertDropImage( {
+				editor: this.editor,
+				event: dropEvt,
+				type: imageType,
+				expected: expected,
+				dropRange: {
+					dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+					dropOffset: 17
+				}
+			} );
+		},
+
+		// (#4750)
+		'test showing notification for dropping unsupported image type': function() {
+			var dropEvt = bender.tools.mockDropEvent(),
+				imageType = 'application/pdf',
+				expected = '<p class="p">Paste image here:</p>',
+				expectedMsg = this.editor.lang.clipboard.fileFormatNotSupportedNotification,
+				expectedDuration = this.editor.config.clipboard_notificationDuration,
+				spy = sinon.spy( this.editor, 'showNotification' );
+
+			FileReader.setFileMockType( imageType );
+			FileReader.setReadResult( 'load' );
+
+			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
+			assertDropImage(  {
+				editor: this.editor,
+				event: dropEvt,
+				type: imageType,
+				expected: expected,
+				dropRange: {
+					dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+					dropOffset: 17
+				},
+				callback: function() {
+					spy.restore();
+
+					assert.areSame( 1, spy.callCount, 'There was only one notification' );
+					assert.areSame( expectedMsg, spy.getCall( 0 ).args[ 0 ],
+						'The notification had correct message' );
+					assert.areSame( 'info', spy.getCall( 0 ).args[ 1 ],
+					'The notification had correct type' );
+					assert.areSame( expectedDuration, spy.getCall( 0 ).args[ 2 ],
```

Comment:
I'm not fully convinced that we should test duration here? I's kind of notification feature and we should trust that this will be handled correctly. WDYT?

Reply:
This test does not check the real duration of the notification, it tests if the argument with intended duration is actually passed to the `editor#showNotifcation()`.

Reply:
Yes, I mean, that we expect that the config option is used to show notification. Because we are giving value from config as expected value. But it doesn't bother me that much :) Thanks for explaining.